src/test/unit/algorithms/ConnectionsTest.cpp

```diff
@@ -348,95 +363,57 @@ TEST(ConnectionsTest, testComputeActivity) {
 
 
 TEST(ConnectionsTest, testAdaptSynapses) {
-  FAIL();
-  /*
   UInt numColumns = 4;
   UInt numInputs = 8;
   Connections con(numColumns);
 
   vector<UInt> activeColumns;
-  vector<UInt> inputVector;
+  SDR input({numInputs});
 
-  UInt potentialArr1[4][8] = {{1, 1, 1, 1, 0, 0, 0, 0},
+  UInt potentialArr[4][8] =  {{1, 1, 1, 1, 0, 0, 0, 0},
                               {1, 0, 0, 0, 1, 1, 0, 1},
                               {0, 0, 1, 0, 0, 0, 1, 0},
                               {1, 0, 0, 0, 0, 0, 1, 0}};
 
-  Real permanencesArr1[5][8] = {
+  Real permanences[4][8] = {
       {0.200, 0.120, 0.090, 0.060, 0.000, 0.000, 0.000, 0.000},
       {0.150, 0.000, 0.000, 0.000, 0.180, 0.120, 0.000, 0.450},
-      {0.000, 0.000, 0.014, 0.000, 0.000, 0.000, 0.110, 0.000},
+      {0.000, 0.000, 0.004, 0.000, 0.000, 0.000, 0.910, 0.000},
       {0.070, 0.000, 0.000, 0.000, 0.000, 0.000, 0.178, 0.000}};
 
-  Real truePermanences1[5][8] = {
+  Real truePerms[4][8] = {
       {0.300, 0.110, 0.080, 0.160, 0.000, 0.000, 0.000, 0.000},
-      //   Inc     Dec   Dec    Inc      -      -      -     -
+      // Inc    Dec    Dec    Inc      -      -      -     -
       {0.250, 0.000, 0.000, 0.000, 0.280, 0.110, 0.000, 0.440},
-      //   Inc      -      -     -      Inc    Dec    -     Dec
-      {0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.210, 0.000},
-      //   -      -     Trim     -     -     -       Inc   -
+      // Inc      -      -     -      Inc    Dec    -     Dec
+      {0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 1.000, 0.000},
+      //   -      -   Floor     -     -     -    Ceiling   -
       {0.070, 0.000, 0.000, 0.000, 0.000, 0.000, 0.178, 0.000}};
-  //    -      -      -      -      -      -      -       -
-
-  UInt inputArr1[8] = {1, 0, 0, 1, 1, 0, 1, 0};
-  UInt activeColumnsArr1[3] = {0, 1, 2};
+      //   -      -      -      -      -      -      -      -
 
   for (UInt column = 0; column < numColumns; column++) {
-    sp.setPotential(column, potentialArr1[column]);
-    sp.setPermanence(column, permanencesArr1[column]);
+    Segment seg = con.createSegment(column);
+    for(UInt inp = 0; inp < numInputs; inp++) {
+      if( potentialArr[column][inp] )
+        con.createSynapse(seg, inp, permanences[column][inp]);
```

Comment:
so this is a new way to create/update permanences on a synapse? If so, are methods setPotential, setPermanence still needed? (or not public at least?)

Reply:
This is the way to use the Connections class.  

setPotential & setPermanence are SpatialPooler methods.

I will clean up these variable names & add comments.